From cc30e0e9fa1260fe04f911634bf880f66923f50d Mon Sep 17 00:00:00 2001
From: William Douglas <william.douglas@intel.com>
Date: Tue, 23 May 2023 10:31:31 -0700
Subject: [PATCH] fileattrs: Don't scan glibc auto-search dirs for Provides

Those libraries are meant to be runtime replacements of the ones in the regular
$libdir, based on CPU features at runtime. If a packager decides to split them into
a separate sub-package, the dependency resolver shouldn't be deciding to install
that instead of the base package.

[2023-05-23]: Ported for rpm 4.18.1 update, account for upstream not
              scanning ko files.

[2021-05-05]: Remove elfoptimized.attr. This will installed by clr-rpm-config
              instead to ease maintenance.

[2021-09-10]: Ported for rpm 4.17.0 update

[2021-10-08]: Squashed all patches to elf.attr into this one; exclude
              new /usr/share/clear/optimized-elf tree as well.

[2022-06-26]: Updated regexp to ignore /usr/lib64/glibc-hwcaps tree

Ported-by: Leandro Dorileo <leandro.maciel.dorileo@intel.com>
Ported-by: Patrick McCarty <patrick.mccarty@intel.com>
Ported-by: William Douglas <william.douglas@intel.com>
Signed-off-by: Thiago Macieira <thiago.macieira@intel.com>
Signed-off-by: Patrick McCarty <patrick.mccarty@intel.com>
Signed-off-by: William Douglas <william.douglas@intel.com>
---
 fileattrs/elf.attr | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fileattrs/elf.attr b/fileattrs/elf.attr
index 177a596..0fbaba1 100644
--- a/fileattrs/elf.attr
+++ b/fileattrs/elf.attr
@@ -1,4 +1,4 @@
 %__elf_provides		%{_rpmconfigdir}/elfdeps --provides
 %__elf_requires		%{_rpmconfigdir}/elfdeps --requires
 %__elf_magic		ELF (32|64)-bit.*$
-%__elf_exclude_path	^/lib/modules/.*\.ko?(\.[[:alnum:]]*)$
+%__elf_exclude_path	^(/lib/modules/.*\.ko?(\.[[:alnum:]]*)$|/usr/lib/debug|/usr/mingw/lib|/usr/share/clear/optimized-elf|/usr/x86_64-generic-linux/x86_64-w64-mingw32/lib|(/usr)?/lib(64)?/(i686|x86_64|sse2|haswell|tls|openmpi|glibc-hwcaps))
-- 
2.40.1

